legal-api/src/legal_api/core/meta/filing.py

```diff
@@ -480,22 +481,49 @@ def get_all_outputs(business_type: str, filing_name: str) -> list:
         return []
 
     @staticmethod
-    def alter_outputs(filing: FilingStorage, outputs: set):
+    def alter_outputs(filing: FilingStorage, business: Business, outputs: set):
         """Add or remove outputs conditionally."""
+        outputs = FilingMeta.alter_outputs_alteration(filing, outputs)
+        outputs = FilingMeta.alter_outputs_correction(filing, business, outputs)
+        outputs = FilingMeta.alter_outputs_special_resolution(filing, outputs)
+        if filing.filing_type == 'dissolution' and filing.filing_sub_type == 'administrative':
+            # Supress Certificate of Dissolution for Admin Dissolution
+            outputs.remove('certificateOfDissolution')
+
+    @staticmethod
+    def alter_outputs_alteration(filing, outputs):
+        """Handle output file list modification for alterations."""
         if filing.filing_type == 'alteration':
             if filing.meta_data.get('alteration', {}).get('toLegalName'):
                 outputs.add('certificateOfNameChange')
-        elif filing.filing_type == 'specialResolution':
+        return outputs
+
+    @staticmethod
+    def alter_outputs_correction(filing, business, outputs):
+        """Handle output file list modification for corrections."""
+        if filing.filing_type == 'correction':
+            corrected_filing_id = filing.filing_json.get('correction', {}).get('correctedFilingId')
+            original_filing = FilingStorage.find_by_id(corrected_filing_id)
```

Comment:
After discussing with Argus - it might be a better practice to just add an `originalFilingType` or an equivalent for corrections into the filing table